packages/next-server/server/next-server.ts

```diff
@@ -441,6 +441,10 @@ export default class Server {
     query: ParsedUrlQuery = {},
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    if (req.method !== 'GET' && req.method !== 'HEAD') {
+      return this.render404(req, res, parsedUrl)
```

Comment:
We shouldn't be returning an explicit `404`, the previous implementation just didn't match -- if we're responding on behalf of the user then it should be a `405`. I still think this should follow the old behavior.